File: Services/CategoryServices/DrivingSchoolApp.Services.CategoryServices.DCategoryAPI/DataAccess/Entities/Lesson.cs
using DrivingSchoolApp.Services.CategoryServices.DCategoryAPI.DataAccess.Entities.Enums;
namespace DrivingSchoolApp.Services.CategoryServices.DCategoryAPI.DataAccess.Entities;

public class Lesson : Base
{
	public LessonTypes Type { get; set; }
	public double Price { get; set; }
	public double TestPrice { get; set; }
	public string Description { get; set; }
	public string LicenceDescription { get; set; }
	public int Duration { get; set; }
	public double? VirtualPracticePrice { get; set; }
	public int? Discount { get; set; }
	public IEnumerable<CategoryLesson> CategoryLessons { get; set; } = new List<CategoryLesson>();
}

File: Frontend/DrivingSchoolWeb/src/app/categories/category-b/models/lesson.ts
import {Base} from 'src/app/categories/shared/models/base';
import {LessonType} from 'src/app/categories/category-b/models/enums/lesson-type';

export interface Lesson extends Base{
  lessonType:LessonType,
  price: number,
  testPrice:number,
  description:string,
  licenceDescription: string,
  duration: number,
  virtualPracticePrice: number | null,
  discount:number | null
}
